lib/node_modules/@stdlib/math/base/complex/divide/lib/cdiv.js

```diff
@@ -0,0 +1,92 @@
+'use strict';
+
+// MODULES //
+
+var FLOAT64_BIGGEST = require( '@stdlib/math/constants/float64-max' );
+var FLOAT64_SMALLEST = require( '@stdlib/math/constants/float64-smallest-normal' );
+var EPS = require( '@stdlib/math/constants/float64-eps' );
+var abs = require( '@stdlib/math/base/special/abs' );
+var max = require( '@stdlib/math/base/special/max' );
+var robustInternal = require( './robust-internal' );
+
+// VARIABLES //
+
+var LARGE_THRESHOLD = FLOAT64_BIGGEST * 0.5;
+var SMALL_THRESHOLD = FLOAT64_SMALLEST * ( 2.0 / EPS );
+var RECIP_EPS_SQR = 2.0 / ( EPS * EPS );
+
+// Create a two-element array in order to pass numbers locally
+// without allocating unnecessary tiny arrays.
+var buffer = [0, 0];
+
+// MAIN //
+
+/**
+* Divides two complex numbers.
+*
+* Based on the algorithm found in:
+*
+* M. Baudin, R. Smith, "A Robust Complex Division in Scilab",
+* https://arxiv.org/pdf/1210.4539.pdf
+*
+* @param {number} re1 - real component
+* @param {number} im1 - imaginary component
+* @param {number} re2 - real component
+* @param {number} im2 - imaginary component
+* @returns {Array<number>} real and imaginary components
+*
+* @example
+* var v = cdiv( -13.0, -1.0, -2.0, 1.0 );
+* // returns [ 5.0, 3.0 ]
+*/
+function cdiv( re1, im1, re2, im2 ) {
+	var ab;
+	var cd;
+	var s;
+	var quotient;
+
+	ab = max(abs(re1), abs(im1));
+	cd = max(abs(re2), abs(im2));
+	s = 1.0;
+
+	if ( ab >= LARGE_THRESHOLD ) {
+		re1 *= 0.5;
+		im1 *= 0.5;
+		s *= 2.0;
+	} else if ( ab <= SMALL_THRESHOLD ) {
+		re1 *= RECIP_EPS_SQR;
+		im1 *= RECIP_EPS_SQR;
+		s /= RECIP_EPS_SQR;
+	}
+
+	if ( cd >= LARGE_THRESHOLD ) {
+		re2 *= 0.5;
+		im2 *= 0.5;
+		s *= 0.5;
+	} else if ( cd <= SMALL_THRESHOLD ) {
+		re2 *= RECIP_EPS_SQR;
+		im2 *= RECIP_EPS_SQR;
+		s *= RECIP_EPS_SQR;
+	}
+
+	quotient = [0, 0];
+
+	if ( abs( im2 ) <= abs( re2 ) ) {
+		robustInternal( buffer, re1, im1, re2, im2 );
```

Comment:
Oh… no good reason. Fixed. Oops. That was a holdover from a previous (crummier) version. Removed.